src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs

```diff
@@ -166,6 +169,47 @@ public sealed partial class ConfidentialClientApplication
                 longRunningProcessSessionKey);
         }
 
+        /// <summary>
+        /// Stops an in progress long running OBO session by removing the tokens associated with the provided cache key.
+        /// </summary>
+        /// <param name="longRunningProcessSessionKey">OBO cache key used to remove the tokens</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="longRunningProcessSessionKey"/> is not set.</exception>
+        public async Task StopLongRunningWebApiAsync(string longRunningProcessSessionKey, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(longRunningProcessSessionKey))
+            {
+                throw new ArgumentNullException(nameof(longRunningProcessSessionKey));
+            }
+
+            Guid correlationId = Guid.NewGuid();
```

Comment:
Why don't we obey the correlation id from the customer passes in?

Reply:
Discussed offline: Remove token apis do not communicate with the endpoint